tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -7164,6 +7213,35 @@ bool SpirvEmitter::isBufferTextureIndexing(const CXXOperatorCallExpr *indexExpr,
   return false;
 }
 
+bool SpirvEmitter::isDescriptorHeap(const Expr *expr) {
+  const CXXOperatorCallExpr *operatorExpr = dyn_cast<CXXOperatorCallExpr>(expr);
+  if (!operatorExpr)
+    return false;
+
+  // Must be operator[]
+  if (operatorExpr->getOperator() != OverloadedOperatorKind::OO_Subscript)
+    return false;
```

Comment:
This one is quite irritating.
DXC handles that by generating a pointer, then forcing legalization knowing "this pointer can be removed from the parameters, and fetched from this builtin".
In the SPIR-V case, it breaks.
I'd definitely want to prevent using this type as template parameter, seems way simpler than hacking around trying to make this work in our codegen.